agent/api/containerstatus.go

```diff
@@ -49,29 +57,58 @@ func (cs ContainerStatus) String() string {
 	return "NONE"
 }
 
-// TaskStatus maps the container status to the corresponding task status
+// TaskStatus maps the container status to the corresponding task status. The
+// transition map is illustreated below.
+//
+// Container: None -> Created -> Running -> Provisioned -> Stopped -> Zombie
+//
+// Task     : None ->     Created        -> Running     -> Stopped
 func (cs *ContainerStatus) TaskStatus() TaskStatus {
 	switch *cs {
 	case ContainerStatusNone:
 		return TaskStatusNone
 	case ContainerCreated:
 		return TaskCreated
 	case ContainerRunning:
+		return TaskCreated
+	case ContainerResourcesProvisioned:
 		return TaskRunning
 	case ContainerStopped:
 		return TaskStopped
 	}
 	return TaskStatusNone
 }
 
-// BackendRecognized returns true if the container status is recognized as a valid state
-// by ECS. Note that not all container statuses are recognized by ECS or map to ECS
-// states
-func (cs *ContainerStatus) BackendRecognized() bool {
-	return *cs == ContainerRunning || *cs == ContainerStopped
+// ShouldReportToBackend returns true if the container status is recognized as a
+// valid state by ECS. Note that not all container statuses are recognized by ECS
+// or map to ECS states
+func (cs *ContainerStatus) ShouldReportToBackend() bool {
+	return *cs == ContainerResourcesProvisioned || *cs == ContainerStopped
+}
+
+// BackendStatus maps the internal container status in the agent to that in the
+// backend
+func (cs *ContainerStatus) BackendStatus() ContainerStatus {
```

Comment:
We weren't sending `ContainerCreated` (`CREATED`) for containers prior to this change as well. This is mostly a refactor of `BackendRecognized` in the previous code.